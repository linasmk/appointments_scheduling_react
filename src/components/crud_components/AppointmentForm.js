/* ========= App Dependencies ============= */
import React from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";

/* ========= Code ============= */
const now = moment();
console.log(now.format("MMM Do, YYYY"));

export default class AppointmentForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      patientName: props.appointment ? props.appointment.patientName : "",
      aptNotes: props.appointment ? props.appointment.aptNotes : "",
      aptDate: props.appointment ? moment(props.appointment.aptDate) : moment(),
      createdAt: props.appointment
        ? moment(props.appointment.createdAt)
        : moment(),
      calenderFocused: false,
      error: "",
    };
  }
  onPatientNameChange = (e) => {
    const patientName = e.target.value;
    this.setState(() => ({
      patientName,
    }));
  };
  onNoteChange = (e) => {
    const aptNotes = e.target.value;
    this.setState(() => ({
      aptNotes,
    }));
  };
  onDateChange = (aptDate) => {
    if (aptDate) {
      this.setState(() => ({ aptDate }));
    }
  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calenderFocused: focused }));
  };
  onSubmit = (e) => {
    e.preventDefault();
    if (!this.state.patientName) {
      this.setState(() => ({
        error: "Patient's name field should not stay empty!",
      }));
    } else {
      this.setState(() => ({
        error: "",
      }));
      this.props.onSubmit({
        patientName: this.state.patientName,
        aptDate: this.state.aptDate.valueOf(),
        createdAt: this.state.createdAt.valueOf(),
        aptNotes: this.state.aptNotes,
      });
      this.setState(() => ({
        patientName: "",
        aptNotes: "",
        aptDate: moment(),
      }));
    }
  };
  render() {
    return (
      <form className="appointment-form" onSubmit={this.onSubmit}>
        {this.state.error && (
          <p className="appointment__form--warning">{this.state.error}</p>
        )}
        <div className="appointment-form__input-block">
          <input
            className="appointment-form__input h-input-styles"
            type="text"
            autoFocus
            placeholder="Add patient's name"
            value={this.state.patientName}
            onChange={this.onPatientNameChange}
          />
          <SingleDatePicker
            date={this.state.aptDate}
            onDateChange={this.onDateChange}
            focused={this.state.calenderFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
            daySize={33}
          />
        </div>
        <div className="appointment-form__textarea-block">
          <textarea
            rows="4"
            className="appointment-form__textarea h-input-styles"
            type="text"
            placeholder="Add notes (optional)"
            value={this.state.aptNotes}
            onChange={this.onNoteChange}
          />
        </div>
        <div className="appointment-form__btn-block">
          <button className="appointment-form__btn h-btn-styles">
            Add Appointment
          </button>
        </div>
      </form>
    );
  }
}
