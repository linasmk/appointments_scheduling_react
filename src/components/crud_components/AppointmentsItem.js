/* ========= App Dependencies ============= */
import React, { useState } from "react";
import { connect } from "react-redux";
import {
  AiOutlineCloseCircle,
  AiFillEdit,
  AiOutlineCheckCircle,
} from "react-icons/ai";
import moment from "moment";

/* ========= Redux ============= */
import {
  editAppointment,
  removeAppointment,
} from "../../store/appointmentsAction";
/* ========= Code ============= */
class AppointmentsItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      id: props.appointment ? props.appointment.id : "",
      patientName: props.appointment ? props.appointment.patientName : "",
      aptNotes: props.appointment ? props.appointment.aptNotes : "",
      aptDate: props.appointment ? moment(props.appointment.aptDate) : moment(),
      createdAt: props.appointment
        ? moment(props.appointment.createdAt)
        : moment(),
      isContentInEditMode: false,
    };
  }
  toggleEditMode = () => {
    this.setState((prevState) => ({
      isContentInEditMode: !prevState.isContentInEditMode,
    }));
  };
  onPatientNameChange = (e) => {
    const patientName = e.target.textContent;
    this.setState(() => ({
      patientName,
    }));
  };
  onAppointmentNotesChange = (e) => {
    const aptNotes = e.target.textContent;
    this.setState(() => ({
      aptNotes,
    }));
  };
  editAppointment = (e) => {
    (appointment) => {
      this.props.dispatch(editAppointment(this.state.id, appointment));
    };
    this.toggleEditMode();
  };
  render() {
    return (
      <section key={this.state.id} className="appointment-item">
        <AiOutlineCloseCircle
          className="appointment-item__close-circle h-icon-styles"
          onClick={(e) => {
            console.log(this.state.patientName);
            this.props.dispatch(removeAppointment({ id: this.state.id }));
          }}
        />
        {this.state.isContentInEditMode ? (
          <AiOutlineCheckCircle
            className="appointment-item__check-circle h-icon-styles"
            onClick={this.editAppointment}
          />
        ) : (
          <AiFillEdit
            className="appointment-item__fill-edit h-icon-styles"
            onClick={this.toggleEditMode}
          />
        )}

        <h3 className="appointment-item__name">
          Patient's name:
          <span
            contentEditable={this.state.isContentInEditMode}
            suppressContentEditableWarning
            onBlur={this.onPatientNameChange}
          >
            {" "}
            {this.state.patientName}
          </span>
        </h3>
        <p className="appointment-item__date">
          Appointment is schedulled for:
          <span
            contentEditable={this.state.isContentInEditMode}
            suppressContentEditableWarning
          >
            {/*this.state.aptDate*/}
          </span>
        </p>
        <h4 className="appointment-item__notes-heading">Notes</h4>
        <p
          className="appointment-item__notes-p"
          contentEditable={this.state.isContentInEditMode}
          suppressContentEditableWarning
          onBlur={this.onAppointmentNotesChange}
        >
          {this.state.aptNotes}
        </p>
        <p className="appointment-item__created-at">
          Created at: <span>{/*this.state.createdAt*/}</span>
        </p>
      </section>
    );
  }
}
// const AppointmentsItem = ({
//   id,
//   patientName,
//   aptNotes,
//   aptDate,
//   createdAt,
//   updates,
//   dispatch,
// }) => {
//   const [isContentInEditMode, setEditMode] = useState(false);

//   const toggleEditMode = () => setEditMode(!isContentInEditMode);
//   return (
//     <section key={id} className="appointment-item">
//       <AiOutlineCloseCircle
//         className="appointment-item__close-circle h-icon-styles"
//         onClick={() => {
//           dispatch(removeAppointment({ id }));
//         }}
//       />
//       {isContentInEditMode ? (
//         <AiOutlineCheckCircle
//           className="appointment-item__check-circle h-icon-styles"
//           onClick={toggleEditMode}
//         />
//       ) : (
//         <AiFillEdit
//           className="appointment-item__fill-edit h-icon-styles"
//           onClick={toggleEditMode}
//         />
//       )}

//       <h3 className="appointment-item__name">
//         Patient's name:{" "}
//         <span
//           contentEditable={isContentInEditMode}
//           suppressContentEditableWarning
//         >
//           {patientName}
//         </span>
//       </h3>
//       <p className="appointment-item__date">
//         Appointment is schedulled for:{" "}
//         <span
//           contentEditable={isContentInEditMode}
//           suppressContentEditableWarning
//         >
//           {aptDate}
//         </span>
//       </p>
//       <h4 className="appointment-item__notes-heading">Notes</h4>
//       <p
//         className="appointment-item__notes-p"
//         contentEditable={isContentInEditMode}
//         suppressContentEditableWarning
//       >
//         {aptNotes}
//       </p>
//       <p className="appointment-item__created-at">
//         Created at: <span>{createdAt}</span>
//       </p>
//     </section>
//   );
// };
const mapStateToProps = (state, props) => {
  return {
    appointment: state.appointments.find(
      (appointment) => appointment.id === props.id
    ),
  };
};

export default connect(mapStateToProps)(AppointmentsItem);
