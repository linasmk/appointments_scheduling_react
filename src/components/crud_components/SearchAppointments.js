/* ========= App Dependencies ============= */
import React, { useState } from "react";
import { connect } from "react-redux";
import { DateRangePicker } from "react-dates";
import { AiFillCaretDown } from "react-icons/ai";
import { BsPlusCircleFill, BsPencilSquare } from "react-icons/bs";
import getVisibleAppointments from "../../store/selectors.js";
import {
  setTextFilter,
  sortByDate,
  sortByPatientsName,
  setStartDate,
  setEndDate,
} from "../../store/filtersAction";
/* ========= Code ============= */
const SearchAppointments = (props) => {
  const [calendarFocused, setState] = useState(null);
  const [isAccordionOpen, setAccordion] = useState(false);

  function onDatesChange({ startDate, endDate }) {
    props.dispatch(setStartDate(startDate));
    props.dispatch(setEndDate(endDate));
  }
  function onFocusChange(calendarFocused) {
    setState(() => calendarFocused);
  }
  const toggleAccordion = () => setAccordion(!isAccordionOpen);

  return (
    <article className="appointments-article">
      <section
        className={isAccordionOpen ? "accordion accordion-open" : "accordion"}
      >
        <header className="accordion__header">
          <h2 className="accordion__heading h-heading-styles">
            Search Appointments
          </h2>
          <BsPlusCircleFill className="plus-circle" onClick={toggleAccordion} />
        </header>
        <div
          className={
            isAccordionOpen
              ? "accordion__body js-accordion-open-search"
              : "accordion__body"
          }
        >
          <div className="searchblock">
            <input
              className="searchblock__input"
              placeholder="Search appointments"
              type="text"
              value={props.filters.text}
              onChange={(e) => {
                props.dispatch(setTextFilter(e.target.value));
              }}
            />
            <div className="searchblock__select-wrapper">
              <select
                className="searchblock__select-btn h-btn-styles"
                value={props.filters.sortBy}
                onChange={(e) => {
                  if (e.target.value === "date") {
                    props.dispatch(sortByDate());
                  } else if (e.target.value === "patient's name") {
                    props.dispatch(sortByPatientsName());
                  }
                }}
              >
                <option value="date">Date</option>
                <option value="patient's name">Name</option>
              </select>
              <AiFillCaretDown className="caret-down" />
            </div>
          </div>

          <div className="datepicker-block">
            <DateRangePicker
              startDate={props.filters.startDate}
              endDate={props.filters.endDate}
              onDatesChange={onDatesChange}
              focusedInput={calendarFocused}
              onFocusChange={onFocusChange}
              startDateId={"dwjkhqkehwqjkeq"}
              endDateId={"cxzvcxbzbczxbz"}
              showClearDates={true}
              numberOfMonths={1}
              isOutsideRange={() => false}
              daySize={33}
            />
          </div>
        </div>
      </section>
    </article>
  );
};

const mapStateToProps = (state) => {
  return {
    filters: state.filters,
  };
};
export default connect(mapStateToProps)(SearchAppointments);
