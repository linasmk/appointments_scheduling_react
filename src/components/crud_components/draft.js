/* ========= App Dependencies ============= */
import React from "react";
import { connect } from "react-redux";
import { DateRangePicker } from "react-dates";
import { AiFillCaretDown } from "react-icons/ai";
import { BsPlusCircleFill } from "react-icons/bs";
import getVisibleAppointments from "../../store/selectors.js";
import {
  setTextFilter,
  sortByDate,
  sortByPatientsName,
  setStartDate,
  setEndDate,
} from "../../store/filtersAction";
/* ========= Code ============= */
class SearchAppointments extends React.Component {
  state = {
    calendarFocused: null,
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({
      calendarFocused,
    }));
  };
  render() {
    return (
      <article className="appointments-article">
        <section className="accordion">
          <header className="accordion__header">
            <h2 className="accordion__heading">Search Appointments</h2>
            <BsPlusCircleFill className="plus-circle" />
          </header>
          <div className="accordion__body">
            <div className="searchblock">
              <input
                className="searchblock__input h-input-styles"
                placeholder="Search appointments"
                type="text"
                value={this.props.filters.text}
                onChange={(e) => {
                  this.props.dispatch(setTextFilter(e.target.value));
                }}
              />
              <div className="searchblock__select-wrapper">
                <select
                  className="searchblock__select-btn"
                  value={this.props.filters.sortBy}
                  onChange={(e) => {
                    if (e.target.value === "date") {
                      this.props.dispatch(sortByDate());
                    } else if (e.target.value === "patient's name") {
                      this.props.dispatch(sortByPatientsName());
                    }
                  }}
                >
                  <option value="date">Date</option>
                  <option value="patient's name">Name</option>
                </select>
                <AiFillCaretDown className="caret-down" />
              </div>
            </div>
            <div></div>
            <div className="datepicker-block">
              <DateRangePicker
                startDate={this.props.filters.startDate}
                endDate={this.props.filters.endDate}
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calendarFocused}
                onFocusChange={this.onFocusChange}
                startDateId={"dwjkhqkehwqjkeq"}
                endDateId={"cxzvcxbzbczxbz"}
                showClearDates={true}
                numberOfMonths={1}
                isOutsideRange={() => false}
                daySize={33}
              />
            </div>
          </div>
        </section>
      </article>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    filters: state.filters,
  };
};
export default connect(mapStateToProps)(SearchAppointments);
