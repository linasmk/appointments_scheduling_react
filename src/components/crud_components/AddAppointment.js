/* ========= Dependencies ============= */
import React, { useState } from "react";
import { connect } from "react-redux";
import { BsPlusCircleFill } from "react-icons/bs";
import { addAppointment } from "../../store/appointmentsAction";

/* ========= Components ============= */
import AppointmentForm from "./AppointmentForm";

/* ========= Code ============= */

export const AddAppointment = (props) => {
  const [isAccordionOpen, setAccordion] = useState(true);
  const toggleAccordion = () => setAccordion(!isAccordionOpen);
  return (
    <article className="appointments-article">
      <section
        className={isAccordionOpen ? "accordion accordion-open" : "accordion"}
      >
        <header className="accordion__header">
          <h2 className="accordion__heading h-heading-styles">
            Add Appointment
          </h2>
          <BsPlusCircleFill className="plus-circle" onClick={toggleAccordion} />
        </header>
        <div
          className={
            isAccordionOpen
              ? "accordion__body js-accordion-open-add"
              : "accordion__body"
          }
        >
          <AppointmentForm
            // onSubmit={(appointment) => {
            //   props.dispatch(addAppointment(appointment));
            // }}
            onSubmit={props.onSubmit}
          />
        </div>
      </section>
    </article>
  );
};

const mapDispatchToProps = (dispatch) => ({
  onSubmit: (appointment) => dispatch(addAppointment(appointment)),
});
export default connect(undefined, mapDispatchToProps)(AddAppointment);
