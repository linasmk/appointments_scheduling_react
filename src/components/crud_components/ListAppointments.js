/* ========== Dependencies ============ */
import React from "react";
/* ========== Redux ============= */
import { connect } from "react-redux";
import getVisibleAppointments from "../../store/selectors";
/* =========== Components ========== */
import AppointmentsItem from "./AppointmentsItem";
/* =========== Code ============= */

const ListAppointments = (props) => (
  <article className="appointments-article">
    <div className="appointments-article__header">
      <h2 className="appointments-article__heading h-heading-styles">
        Appointments List
      </h2>
    </div>
    {props.appointments.map((appointment) => {
      return <AppointmentsItem key={appointment.id} {...appointment} />;
    })}
  </article>
);

const mapStateToProps = (state) => {
  return {
    appointments: getVisibleAppointments(state.appointments, state.filters),
  };
};

export default connect(mapStateToProps)(ListAppointments);
