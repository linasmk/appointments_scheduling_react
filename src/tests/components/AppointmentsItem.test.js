/* ========= App Dependencies ============= */
import React from "react";
import { shallow } from "enzyme";
import { AppointmentsItem } from "../../components/crud_components/AppointmentsItem";
import appointments from "../fixtures/appointments";
/* ========= Code ============= */

test("Should render AppointmentsItem correctly", () => {
  const wrapper = shallow(<AppointmentsItem {...appointments[0]} />);
  expect(wrapper).toMatchSnapshot();
});
