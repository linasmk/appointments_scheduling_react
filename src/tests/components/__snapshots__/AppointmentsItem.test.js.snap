// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render AppointmentsItem correctly 1`] = `
<section
  className="appointment-item"
  key=""
>
  <AiOutlineCloseCircle
    className="appointment-item__close-circle h-icon-styles"
    onClick={[Function]}
  />
  <AiFillEdit
    className="appointment-item__fill-edit h-icon-styles"
    onClick={[Function]}
  />
  <h3
    className="appointment-item__name"
  >
    Patient's name:
    <span
      contentEditable={false}
      onBlur={[Function]}
      suppressContentEditableWarning={true}
    >
       
    </span>
  </h3>
  <div
    className="appointment-item__date"
  >
    <p>
      Appointment is schedulled for: Â 
    </p>
    <p>
      January 1st, 1970. At: 1:00 AM
    </p>
  </div>
  <h4
    className="appointment-item__notes-heading"
  >
    Notes
  </h4>
  <p
    className="appointment-item__notes-p"
    contentEditable={false}
    onBlur={[Function]}
    suppressContentEditableWarning={true}
  />
  <p
    className="appointment-item__created-at"
  >
    Created at:
     
    <span>
      January 1st, 1970. At: 1:00 AM
    </span>
  </p>
</section>
`;
